Pod/Classes/SeafDateFormatter.m

```diff
@@ -62,45 +72,39 @@ + (NSString *)stringFromLongLong:(long long)time
         return [[SeafDateFormatter sharedLoader] stringFromDate:date];
 }
 
-+(NSString *)compareCurrentFromGMTDate:(NSString *)timeStr {
-    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
-    [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ssZ"];
-    NSTimeZone *localTimeZone = [NSTimeZone localTimeZone];
-    [dateFormatter setTimeZone:localTimeZone];
-    NSDate *dateFormatted = [dateFormatter dateFromString:timeStr];
-
-    [dateFormatter setDateFormat:@"yyyy-MM-dd HH:mm:ss"];
-    NSDate *timeDate = [dateFormatter dateFromString:[dateFormatter stringFromDate:dateFormatted]];
-    NSDate *currentDate = [NSDate date];
++(NSString *)compareGMTTimeWithNow:(NSString *)gmtTimeStr {
+    NSDate *dateFormatted = [[SeafDateFormatter sharedLoaderUTC] dateFromString:gmtTimeStr];
+    NSTimeInterval timeInterval = [[NSDate date] timeIntervalSinceDate:dateFormatted];
+    
+    double minutes = timeInterval / 60;
+    double hours = minutes / 60;
+    double days = hours / 24;
+    double months = days / 30;
+    double years = months / 12;
     
-    NSTimeInterval timeInterval = [currentDate timeIntervalSinceDate:timeDate];
-    long temp = 0;
-    NSString *result;
-    if (timeInterval/60 < 1) {
-        result = NSLocalizedString(@"a few seconds ago", @"Seafile");
-    } else if (timeInterval/60 == 1) {
-        result = NSLocalizedString(@"a minute ago", @"Seafile");
-    } else if ((temp = timeInterval/60) < 60 ) {
-        result = [NSString stringWithFormat:NSLocalizedString(@"%ld minutes ago", @"Seafile"), temp];
-    } else if ((temp = timeInterval/60) == 60) {
-        result = [NSString stringWithFormat:NSLocalizedString(@"an hour ago", @"Seafile"), temp];
-    } else if ((temp = timeInterval/3600) < 24 ) {
-        result = [NSString stringWithFormat:NSLocalizedString(@"%ld hours ago", @"Seafile"), temp];
-    } else if ((temp = timeInterval/3600) == 24 ) {
-        result = NSLocalizedString(@"a day ago", @"Seafile");
-    } else if((temp = timeInterval/(3600*24)) < 30 ) {
-        result = [NSString stringWithFormat:NSLocalizedString(@"%ld days ago", @"Seafile"), temp];
-    } else if((temp = timeInterval/(3600*24)) == 30 ) {
-        result = [NSString stringWithFormat:NSLocalizedString(@"a month ago", @"Seafile"), temp];
-    } else if((temp = timeInterval/(3600*24*30)) < 12 ) {
-        result = [NSString stringWithFormat:NSLocalizedString(@"%ld months ago", @"Seafile"), temp];
-    } else if((temp = timeInterval/(3600*24*30)) == 12 ) {
-        result = [NSString stringWithFormat:NSLocalizedString(@"a year ago", @"Seafile"), temp];
+    if (minutes < 1.0) {
+        return NSLocalizedString(@"a few seconds ago", @"Seafile");
+    } else if (minutes == 1.0) {
```

Comment:
For float and double, you can't do equal comparison. https://stackoverflow.com/questions/9850663/comparing-same-float-values-in-c